### PR TITLE
FIX: forward link edges + allow for stdin usage in `whatrecord [parse|graph]`

### PR DESCRIPTION
**Pull request by klauer**
## Description
Fix FLNK source ports and direction.

## Motivation and Context
Forward link edges were inaccurately drawn with graphviz, so this PR should fix them.

## Screenshots (if appropriate):
Looks correct now:
image

### CODE REVIEW

File: whatrecord/parse.py [hunk 1]
```diff
@@ -74,7 +75,20 @@ def parse(
...
-    filename = pathlib.Path(filename)
+    if str(filename) == "-":
+        contents = sys.stdin.read()
+        filename = pathlib.Path("/dev/stdin")
+
+        if format is None:
+            options = [fmt.name for fmt in list(FileFormat)]
+            raise ValueError(
+                f"Format must be specified when piping data from standard "
+                f"input.  Choose one of: {options}"
```
Comment (ZLLentz) [inline whatrecord/parse.py:86]: Maybe this check should come before you read from stdin

File: whatrecord/parse.py [hunk 2]
```diff
@@ -74,7 +75,20 @@ def parse(
     """
     standin_directories = standin_directories or {}
 
-    filename = pathlib.Path(filename)
+    if str(filename) == "-":
+        contents = sys.stdin.read()
+        filename = pathlib.Path("/dev/stdin")
```
Comment (tangkong) [inline whatrecord/parse.py:80]: I suppose this is unix-only then?  I'm not sure how to make it work on windows, but I'm sure it's not too big a deal for now.

**Review by tangkong — APPROVED** [review]
LGTM.  👍

File: whatrecord/parse.py [hunk 3]
```diff
@@ -84,17 +98,26 @@ def parse(
...
             dbd=Database.from_file(dbd, version=3 if v3 else 4),
             macro_context=macro_context
         )
 
     if format == FileFormat.iocsh:
-        md = IocMetadata.from_file(
+        if filename == pathlib.Path("/dev/stdin"):
+            raise ValueError(
+                "IOC shell script parsing not supported through standard input"
+            )
```
Comment (ZLLentz) [inline whatrecord/parse.py:118]: Can this be merged with the previous file format checks in the earlier block? fail faster and all that jazz

**Review by ZLLentz — APPROVED** [review]
Small suggestions and a green check

I never noticed that the arrows were inconsistent

File: whatrecord/parse.py (hunk 1 above)
Comment (klauer) [reply whatrecord/parse.py:86]: Good call; addressed in d843fd5

File: whatrecord/parse.py (hunk 2 above)
Comment (klauer) [reply whatrecord/parse.py:80]: The `/dev/stdin` isn't so meaningful, more of a user-facing name to use for context: e.g., "in `/dev/stdin` on line 5, there was a linter warning"

Your comment made me think I should be using something like `sys.stdin.name` instead, though. That shows up as `<stdin>` on macOS, at least.

Addressed in d843fd5

File: whatrecord/parse.py (hunk 3 above)
Comment (klauer) [reply whatrecord/parse.py]: Very logical and agreed - addressed in d843fd5

**Pull request merged**